Makefile

```diff
@@ -1,47 +1,91 @@
-		| entr -d make lint vagrant_up; \
+		find defaults/ meta/ tasks/ templates/ tests/test.yml tests/Vagrantfile \
+		| entr -d make lint vagrant; \
 	done
 
-test: test_deps vagrant_up
+## Run tests
+test: test_deps lint vagrant
+
+## ! Executes Ansible tests using local connection
+# run it ONLY from within a test VM.
+# Example: make test_ansible
+#          make test_ansible TEST_PLAYBOOK=test-something-else.yml
+test_ansible: test_ansible_build test_ansible_configure
+	cd tests && ansible-playbook \
+		--inventory inventory \
+		--connection local \
+		--tags assert \
+		$(TEST_PLAYBOOK)
 
-integration_test: clean integration_test_deps vagrant_up clean
+## ! Executes Ansible tests using local connection
+# run it ONLY from witinh a test VM.
+# Example: make test_ansible_build
+#          make test_ansible_build TEST_PLAYBOOK=test-something-else.yml
+test_ansible_%:
+	cd tests && ansible-playbook \
+		--inventory inventory \
+		--connection local \
+		--tags=$(subst test_ansible_,,$@) \
+		$(TEST_PLAYBOOK)
+	cd tests && ansible-playbook \
+		--inventory inventory \
+		--connection local \
+		--tags=$(subst test_ansible_,,$@) \
+		$(TEST_PLAYBOOK) \
+		| grep -q 'changed=0.*failed=0' \
+			&& (echo 'Idempotence test: pass' && exit 0) \
+			|| (echo 'Idempotence test: fail' && exit 1)
 
+## Install test dependencies
 test_deps:
-	rm -rf tests/vagrant/sansible.*
-	ln -s ../.. tests/vagrant/sansible.postgresql
+	rm -rf tests/sansible.*
+	ln -s .. tests/sansible.postgresql
+
+## Start and (re)provisiom Vagrant test box
+vagrant:
+	cd tests && vagrant up --no-provision
+	cd tests && vagrant provision
+	@echo "- - - - - - - - - - - - - - - - - - - - - - -"
+	@echo "           Provisioning Successful"
+	@echo "- - - - - - - - - - - - - - - - - - - - - - -"
 
-integration_test_deps:
-	sed -i.bak \
-		-E 's/(.*)version: (.*)/\1version: origin\/$(BRANCH)/' \
-		tests/vagrant/integration_requirements.yml
-	rm -rf tests/vagrant/sansible.*
-	ansible-galaxy install -p tests/vagrant -r tests/vagrant/integration_requirements.yml
-	mv tests/vagrant/integration_requirements.yml.bak tests/vagrant/integration_requirements.yml
+## Execute simple Vagrant command
+# Example: make vagrant_ssh
+#          make vagrant_halt
+vagrant_%:
+	cd tests && vagrant $(subst vagrant_,,$@)
 
+## Lint role
+# You need to install ansible-lint
 lint:
-	! find handlers/ meta/ tasks/ -name "*.yml" -type f | xargs grep -E "({{[^ ]|[^ ]}})"
-
-vagrant_up:
-	@cd tests/vagrant; \
-	if (vagrant status | grep -E "(running|saved|poweroff)" 1>/dev/null) then \
-		vagrant up || exit 1; \
-		vagrant provision || exit 1; \
-	else \
-		vagrant up || exit 1; \
-	fi;
-
-vagrant_ssh:
-	@cd tests/vagrant; \
-	vagrant up || exit 1; \
-	vagrant ssh
+	find defaults/ meta/ tasks/ templates/ -name "*.yml" | xargs -I{} ansible-lint {}
 
+## Clean up
 clean:
-	rm -rf tests/vagrant/sansible.*
+	rm -rf tests/sansible.*
 	cd tests/vagrant && vagrant destroy
```

Comment:
The `tests/vagrant` target does not exist. This line should read `cd tests && vagrant destroy` or even better, if you think it could be desired, `cd tests && vagrant destroy -f` to suppress the `default: Are you sure you want to destroy the 'default' VM?` message.